servers/rendering/rendering_server_default.cpp

```diff
@@ -194,6 +184,23 @@ void RenderingServerDefault::_draw(bool p_swap_buffers, double frame_step) {
 	RSG::utilities->update_memory_info();
 }
 
+void RenderingServerDefault::_run_post_draw_steps() {
```

Comment:
I am not sure I understand this, the idea is that this runs in the same thread as the render code, is it not?
I think this should be deprecated though to instead rely on call_on_render_thread()

Reply:
The lines I moved to this function will only work on the main thread, because they are emitting a signal and calling some callbacks. Both mechanisms are only safe on the main thread.

Reply:
That said, I can't see why we have two mechanisms for quite the same. The signal should be enough. But, still, it's only safe to emit it from the main thread. In separate thread mode that's the best point I found to emit it, but there may be better options.

So, deprecating frame drawn callbacks is certainly something we can do, but they still need to be called from the main thread.